src/libraries/System.Text.RegularExpressions/tests/RegexCultureTests.cs

```diff
@@ -3,14 +3,50 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Globalization;
+using System.Linq;
 using System.Tests;
-using Microsoft.DotNet.RemoteExecutor;
 using Xunit;
 
 namespace System.Text.RegularExpressions.Tests
 {
     public class RegexCultureTests
     {
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.IgnoreCase | RegexOptions.CultureInvariant)]
+        [InlineData(RegexOptions.Compiled)]
+        [InlineData(RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant)]
+        public void CharactersComparedOneByOne(RegexOptions options)
+        {
+            // Regex compares characters one by one.  If that changes, it could impact the behavior of
+            // a case like this, where these characters are not the same, but the strings compare
+            // as equal with the invariant culture (and some other cultures as well).
+            const string S1 = "\u00D6\u200D";
+            const string S2 = "\u004F\u0308";
+
+            Assert.False(S1[0] == S2[0]);
+            Assert.False(S1[1] == S2[1]);
+            Assert.StartsWith(S1, S2, StringComparison.InvariantCulture);
+            Assert.True(S1.Equals(S1, StringComparison.InvariantCulture));
```

Comment:
I could be misunderstanding the intent, but I believe this should be `S1.Equals(S2` 

Also, I really am enjoying all the regex perf optimizations you are tackling @stephentoub

Reply:
Oops! Thanks, will fix.